src/irc/ConnectionInstance.ts

```diff
@@ -382,8 +384,11 @@ export class ConnectionInstance {
             retryCount: 0,
             family: server.getIpv6Prefix() || server.getIpv6Only() ? 6 : null,
             bustRfc3484: true,
-            sasl: opts.password ? server.useSasl() : false,
```

Comment:
one concern that a blank password can now be incorrectly used with sasl. We should probably catch and throw this?